### PR TITLE
feat(auth): implement mass assignment vulnerability

### PR DESCRIPTION
**Pull request by Kif-Fando**

### CODE REVIEW

File: src/users/users.controller.ts [hunk 1]
```diff
@@ -260,6 +259,41 @@ export class UsersController {
...
+  })
+  @ApiOkResponse({
+    description: 'Returns updated user',
+  })
+  async changeUserInfo(
+    @Body() body: User,
+    @Param('email') email: string,
+  ) {
+    try {
+      const newInfo = body;
```
Comment (derevnjuk) [inline src/users/users.controller.ts:287]: Local variable is redundant here

File: src/users/users.service.ts [hunk 2]
```diff
@@ -56,6 +56,19 @@ export class UsersService {
     return user;
   }
 
+  async updateUserInfo(email: string, info: User): Promise<User> {
+    this.log.debug(`updateUserInfo ${email}`);
+    const user = await this.findByEmail(email);
+    if (!user) {
+      throw new Error('Could not find user');
```
Comment (derevnjuk) [inline src/users/users.service.ts:63]: Why don't raise 404 HTTP exception in this case?

**Review by derevnjuk — CHANGES_REQUESTED** [review]

File: src/users/users.service.ts [hunk 3]
```diff
@@ -9,8 +9,8 @@ export class UsersService {
   public static readonly LDAP_SEARCH_QUERY = (email) =>
     `(&(objectClass=person)(objectClass=user)(email=${email}))`;
   public static readonly LDAP_ERROR_RESPONSE = `
-      Lookup failed: javax.naming.NamingException: 
-      [LDAP: error code 1 - 000004DC: Lda pErr: DSID-0C0906DC, comment: context not found., data 0, v1db1 ]; 
+      Lookup failed: javax.naming.NamingException:
+      [LDAP: error code 1 - 000004DC: Lda pErr: DSID-0C0906DC, comment: context not found., data 0, v1db1 ];
```
[the remark this answers is not in the document]
Comment (Kif-Fando) [reply src/users/users.service.ts:13]: Tell it to prettier, it doesn`t like these spaces ;)

File: src/users/users.controller.ts [hunk 4]
```diff
@@ -260,6 +259,41 @@ export class UsersController {
...
+        message: { type: 'string' },
+        error: { type: 'string' },
+      },
+    },
+  })
+  @ApiOkResponse({
+    description: 'Returns updated user',
+  })
+  async changeUserInfo(
+    @Body() body: User,
```
[the remark this answers is not in the document]
Comment (Kif-Fando) [reply src/users/users.controller.ts:283]: My fault, fixed

**Review by derevnjuk — APPROVED** [review]

**Pull request merged**